frontend2/src/components/compete/ScrimmagingRecord.tsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import {
+  CompeteMatchScrimmagingRecordRetrieveScrimmageTypeEnum,
+  type TeamPublic,
+} from "api/_autogen";
+import { useScrimmagingRecord } from "api/compete/useCompete";
+import { useEpisodeId } from "contexts/EpisodeContext";
+import WinLossTie from "./WinLossTie";
+
+interface ScrimmagingRecordProps {
+  team: Pick<TeamPublic, "id" | "name" | "profile">;
+  hideTeamName?: boolean;
+  hideAllScrimmages?: boolean;
+  hideUnranked?: boolean;
+  hideRanked?: boolean;
+  className?: string;
+}
+
+const ScrimmagingRecord: React.FC<ScrimmagingRecordProps> = ({
+  team,
+  hideTeamName = false,
+  hideAllScrimmages = false,
+  hideUnranked = false,
+  hideRanked = false,
+  className = "",
+}) => {
+  const { episodeId } = useEpisodeId();
+  const scrimmagingRecordAll = useScrimmagingRecord({
+    episodeId,
+    teamId: team.id,
+    scrimmageType: CompeteMatchScrimmagingRecordRetrieveScrimmageTypeEnum.All,
+  });
+  const scrimmagingRecordUnranked = useScrimmagingRecord({
+    episodeId,
+    teamId: team.id,
+    scrimmageType:
+      CompeteMatchScrimmagingRecordRetrieveScrimmageTypeEnum.Unranked,
+  });
+  const scrimmagingRecordRanked = useScrimmagingRecord({
+    episodeId,
+    teamId: team.id,
+    scrimmageType:
+      CompeteMatchScrimmagingRecordRetrieveScrimmageTypeEnum.Ranked,
+  });
```

Comment:
These are bits of repeated code that could be pushed into the WinLossTie component by having that component take the scrimmage type.

Reply:
Done!